src/main/java/uk/gov/pay/directdebit/payments/params/PaymentViewSearchParams.java

```diff
@@ -72,14 +109,25 @@ public PaginationParams getPaginationParams() {
     }
 
     public SearchDateParams getSearchDateParams() { return searchDateParams; }
+    
+    public PaymentViewSearchParams withPage(Long page) {
+        this.page = page;
+        return this;
+    }
 
     public String generateQuery() {
         StringBuilder sb = new StringBuilder("");
-        if (searchDateParams.getFromDate() != null) {
-            sb.append(" AND t.created_date > :" + FROM_DATE_FIELD);
+        if (isNotBlank(mandateId)) {
+            sb.append(" AND m.external_id = :" + MANDATE_EXTERNAL_ID_FIELD);
+            return sb.toString();
```

Comment:
can you explain this? I don't think I understand why we ignore all the other params if mandate id set?

Reply:
In the requirement we explicitly say mandate id (it was not added as an extra search param to PP-3646). It was originally thought to be a different search to that of transactions search. I combined the two as both should return the same search response. At the kick off time it was decided that we ignore the other search results for now. It was also discussed that ignoring other params feature is going to be removed at a later stage.

Reply:
I'm not convinced by this, and I think it could become rather confusing. I think there is a mixing up of different concepts here: 
- the idea of payments as a subresource of mandate (i.e. `/mandate/:mandateId/payments)
- the idea of a search for payments matching certain query params (i.e. `/payments?mandate=:mandateId`)
You seem to be implementing the first concept using the tools of the second concept. I don't think I am qualified to decide if this is the way to go in this project (I personally don't think it is). I realise in publicapi there has been a deliberate decision to do it this way, but I am not sure that means we need to persist it all the way through

Reply:
I don't know what to say to that.